### PR TITLE
Update `tests/tuner/*.py` to use `devices` instead of `gpus` or `ipus`

### PR DESCRIPTION
**Pull request by mathemusician**
## What does this PR do?

Update test to deprecate `num_processes`, `gpus`, `tpu_cores`, and `ipus` from the Trainer constructor

Part of [Pull 11040](https://github.com/PyTorchLightning/pytorch-lightning/pull/11040).

🌳 [Trello Board Link](https://trello.com/b/He49BfCc/pl-tests)

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: tests/tuner/test_auto_gpu_select.py [hunk 1]
```diff
@@ -16,31 +16,11 @@
...
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from tests.helpers.runif import RunIf
 
 
-@RunIf(min_gpus=2)
-@pytest.mark.parametrize(
-    ["auto_select_gpus", "gpus", "expected_error"],
-    [(True, 0, MisconfigurationException), (True, -1, None), (False, 0, None), (False, -1, None)],
-)
-def test_trainer_with_gpus_options_combination_at_available_gpus_env(auto_select_gpus, gpus, expected_error):
```
Comment (carmocca) [inline tests/tuner/test_auto_gpu_select.py:30]: Why is this test removed?

I guess the message needs to be updated to work with the new API

cc @kaushikb11

Comment (mathemusician) [reply tests/tuner/test_auto_gpu_select.py:30]: Yes! So `auto_select_gpus` doesn't really work with `devices` and `accelerator`.

Comment (mathemusician) [reply tests/tuner/test_auto_gpu_select.py:30]: Alternatively, I could wrap the test with `pytest.deprecated_call()`

Comment (kaushikb11) [reply tests/tuner/test_auto_gpu_select.py:30]: > auto_select_gpus doesn't really work with devices and accelerator. 

Why wouldn't it?

Comment (carmocca) [reply tests/tuner/test_auto_gpu_select.py:30]: I would go with the `deprecated_call` and add a TODO about updating the messaging to mention the new arguments

Comment (mathemusician) [reply tests/tuner/test_auto_gpu_select.py:30]: @kaushikb11 That is exactly my question. I’m going to use Paperspace for the next couple of pull requests to help debug faster. Hopefully that speeds up the commits/responses. If anyone knows how to mock multiple gpus, please let me know.

Comment (tchaton) [reply tests/tuner/test_auto_gpu_select.py:30]: Hey @mathemusician. I am curious, what are the advantages of using PaperSpace you don't find elsewhere?

Comment (carmocca) [reply tests/tuner/test_auto_gpu_select.py:30]: > If anyone knows how to mock multiple gpus, please let me know.

You can find multiple examples of exactly that in this file:

`$ grep -iirn "mock" tests/accelerators/test_accelerator_connector.py `

Usually, just `device_count` works, but depends on the test

Comment (mathemusician) [reply tests/tuner/test_auto_gpu_select.py:30]: @tchaton I may just be ignorant about the other options, but it was the first thing I saw that I could ssh tunnel into and tweak code while running multiple gpus.

**Review by awaelchli — APPROVED** [review]

File: tests/tuner/test_auto_gpu_select.py (hunk 1 above)
Comment (mathemusician) [reply tests/tuner/test_auto_gpu_select.py:30]: I decided to put back the test. I added a TODO for @daniellepintz

Comment (mathemusician) [reply tests/tuner/test_auto_gpu_select.py:30]: > grep -iIrn "mock" tests/accelerators/test_accelerator_connector.py

Also that is so cool. I'm like learning so much about `grep` just by doing this.

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by daniellepintz** [comment]
Can someone merge this please?

**Pull request merged**